### PR TITLE
Setup deps.edn build and deploy

### PR DESCRIPTION
**Pull request by kongeor**
Per our discussion:

- switching to deps.edn
- CI with github actions
- adding clj-kondo linting
- building and deploying with tools build

Addresses #24 and #26

### CODE REVIEW

**Comment by kongeor** [comment]
The way tools build is wired it will follow the versioning scheme of `MAJOR.MINOR.COMMITS`. With this library being stable enough, this will look something like `1.0.85` etc. I like this versioning scheme, but if you have a different preference, let me know so we can adjust the code accordingly.

I have also published a test version which is the `1.0.999-SNAPSHOT`.

**Review by hlship — APPROVED** [review]
I'll give these a try and deploy a snapshot version.

**Comment by hlship** [comment]
I still need @trhura to add me to the Clojars group so that i can make a deployment.

File: build.clj [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  (b/delete {:path "target"})
+  (b/delete {:path "cljs-test-runner-out"}))
+
+(defn jar [opts]
+  (b/write-pom {:class-dir class-dir
+                :lib       lib
+                :version   (if (:snapshot opts) snapshot version)
+                :basis     basis
+                :src-dirs  ["src"]})
+  (b/copy-dir {:src-dirs   ["src" "resources"]
```
Comment (hlship) [inline build.clj:22]: This project doesn't (currently) have a `resources` directory.

**Review by hlship — COMMENTED** [review]
Additionally, it would be nice to generate `codox` output and host it; I have a github pages project for my other open source projects that would work.

Finally, if I had done this myself, I would have started with https://github.com/hlship/build-tools however, I've only used that for my other libs and never documented anything, though it is straight forward.

**Comment by trhura** [comment]
@hlship  clojure-humanize project is maintained by @kongeor in Clojars. I don't see it under my projects to add you as a member. 

See https://clojars.org/clojure-humanize

**Comment by kongeor** [comment]
Hm, I'm not an admin of this project. A user with the handle `nickzam` is.

**Comment by kongeor** [comment]
See #1

File: deps.edn [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{:deps {org.clojure/clojure {:mvn/version "1.11.1"}
+        org.clojure/math.numeric-tower {:mvn/version "0.0.5"}
+        clj-time/clj-time {:mvn/version "0.15.2"}
+        com.andrewmcveigh/cljs-time {:mvn/version "0.5.2"}}
+ :paths ["src" "test"]
```
Comment (danielcompton) [inline deps.edn:5]: If `"test"` is included here, it will be shipped in the JAR/gitlib to everyone.

```suggestion
 :paths ["src"]
```

Comment (kongeor) [reply deps.edn]: Thank you for spotting this. Fixed.

**Comment by kongeor** [comment]
I couldn't find the user `nickzam` so this puts the clojars permission issue in a weird state. I'm not sure what is optimal here. I would consider changing the group of the project to something someone of us has admin rights and then grant admin privs to the other 2 for redundancy. It's inconvenient but I can't think of something else. Please let me know what you think.

**Comment by danielcompton** [comment]
@kongeor, @trhura, I talked with @tobias about what to do here.

Given the context around @nickzam creating the group in #1 even though Nick wasn't the "owner" of the code, we feel happy about converting @kongeor from a member of the group to being an admin. Then @kongeor will be able to manage the group members from there.

How does that sound?

**Comment by nickzam** [comment]
Hi guys, are there something I can help you with?

**Comment by trhura** [comment]
@nickzam  Please help add @kongeor @hlship to https://clojars.org/clojure-humanize project as admins.

**Comment by nickzam** [comment]
Done.

**Comment by kongeor** [comment]
Thank you all for helping with this.

@hlship I've tested a snapshot version and it looks all right. You can try it too, if you want to verify that everything works as expected, and then we can release it :)

**Comment by hlship** [comment]
We're currently looking at moving this under the clj-commons umbrella; neither I nor @kongeor have the admin rights to do so.  When you get a chance, could you add one of us?

**Pull request merged**